crates/virtio-queue/src/mock.rs

```diff
@@ -373,14 +382,13 @@ impl<'a, M: GuestMemory> MockSplitQueue<'a, M> {
 
     /// Writes a single descriptor chain to the memory object of the queue, at the beginning of the
     /// descriptor table, and returns the associated `DescriptorChain` object.
-    pub fn build_desc_chain(&self, descs: &[Descriptor]) -> DescriptorChain<&M> {
-        self.add_desc_chain(descs, 0);
-
+    pub fn build_desc_chain(&self, descs: &[Descriptor]) -> Result<DescriptorChain<&M>, Error> {
+        self.add_desc_chain(descs, 0)?;
         self.create_queue(self.mem)
             .iter()
             .unwrap()
             .next()
-            .expect("failed to build desc chain")
+            .ok_or(Error::InvalidDescriptorIndex)
```

Comment:
Does `next` fail only in case of an invalid index? Also I think we are iterating over descriptor chains, so I'm not sure if `Descriptor` is correct there.

Reply:
I added an InvalidNextAvail Error because I saw that the next function fails because of next_avail